### PR TITLE
catbuffer README is outdated

### LINKED ISSUE #55: catbuffer README is outdated
As commented by @Vektrat on Slack, "[basic commands are not working](https://nemgroup-workspace.slack.com/archives/C0153NK559C/p1601192413004400?thread_ts=1601185560.003900&cid=C0153NK559C)", referring to `README.md`.

I can take a look at it myself, at some point. Feel free to add me to the project and assign this issue to me.

### PR DESCRIPTION
**Pull request by segfaultxavi**
Since catbuffer and catbuffer-generators are separate
repos, the commands in the README were not
accurate.
Fixes #55

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,28 +40,35 @@ python main.py [OPTIONS]
...
+```bash
+PYTHONPATH=. python3 catbuffer/main.py --schema catbuffer/schemas/transfer/transfer.cats --include catbuffer/schemas --generator cpp_builder
 ```
-python main.py --schema schemas/transfer/transfer.cats --generator cpp_builder
-```
-You can also generate code for all the schemas running the following command under the ``catbuffer-generators`` directory:
 
-```
-python ../scripts/generate_all.sh cpp_builder
+However, it is much more convenient to use the ``catbuffer-generators`` helper scripts. For instance, you can generate C++ code for all the schemas running the following command (from the ``catbuffer-generators`` folder):
```
Comment (Jaguar0625) [inline README.md:49]: by `catbuffer-generators` do you mean `catbuffer`? 

scripts in this repo should run without needing generators project.

e.g. 
```
bash ./scripts/generate_all.sh cpp_builder /catapult
```

File: README.md [hunk 2]
```diff
@@ -40,28 +40,35 @@ python main.py [OPTIONS]
...
+
+```bash
+scripts/generate_all.sh cpp_builder
```
Comment (Jaguar0625) [inline README.md:52]: i would note somewhere if not already that scripts require bash v5+

File: README.md [hunk 3]
```diff
@@ -40,28 +40,35 @@ python main.py [OPTIONS]
...
 ```
 
 The generator creates a new file for every schema under the ``catbuffer/_generated/cpp_builder`` folder.
 
 ### Run the linter
-```
-pylint --load-plugins pylint_quotes main.py catparser generators test
+
+From the ``catbuffer-generators`` folder:
```
Comment (Jaguar0625) [inline README.md:59]: i'm not sure why this readme is referring to another project?

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

**Comment by segfaultxavi** [comment]
The existing installation instructions said that `catbuffer` must be cloned inside `catbuffer-generators` (as if it was a git submodule) and I continued with that.

I can make the instructions for `catbuffer` completely independent from `catbuffer-generators`, and then add instructions to the later (whose README is currently empty). Would this approach work? Is `catbuffer` really usable without the other project?

**Comment by Jaguar0625** [comment]
You need catbuffer-generators project for first example
```
PYTHONPATH=. python3 catbuffer/main.py --schema catbuffer/schemas/transfer/transfer.cats --include catbuffer/schemas --generator cpp_builder
```

But none of the other examples should require it.

File: README.md (hunk 2 above)
Comment (segfaultxavi) [reply README.md]: My bash is 4.4 and it seems to be working fine. Are you sure we need to include this comment?

File: README.md (hunk 1 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -2,66 +2,76 @@
 
 [![Build Status](https://api.travis-ci.com/nemtech/catbuffer.svg?branch=main)](https://travis-ci.com/nemtech/catbuffer)
 
-The catbuffer library defines the protocol to serialize and deserialize Symbol entities.
+The catbuffer library defines the protocol to serialize and deserialize Symbol entities. Code generators from the [catbuffer-generators](https://github.com/nemtech/catbuffer-generators) project can then produce the leanest code necessary to serialize and deserialize those entities.
 
-In combination with the [catbuffer-generators](https://github.com/nemtech/catbuffer-generators) project, developers can generate builder classes for a given set of programming languages. For example, the [Symbol SDKs](https://nemtech.github.io/sdk) use the generated code to operate with the entities in binary form before announcing them to the network.
+Using catbuffer-generators developers can generate builder classes for a given set of programming languages. For example, the [Symbol SDKs](https://nemtech.github.io/sdk) use the generated code to operate with the entities in binary form before announcing them to the network.
```
Comment (Jaguar0625) [inline README.md:7]: comma after generators

Comment (Jaguar0625) [reply README.md:7]: operate => interact/manipulate

File: README.md [hunk 5]
```diff
@@ -2,66 +2,76 @@
...
+| -s, --schema TEXT    | Input CATS file                                                                                                                                            |               |
+| -o, --output TEXT    | Output directory                                                                                                                                           | _generated    |
+| -i, --include TEXT   | Schema root directory                                                                                                                                      | ./schemas     |
+| -g, --generator TEXT | Generator to use to produce output files (see the [available generators](https://github.com/nemtech/catbuffer-generators/blob/main/generators/All.py#L4)). |               |
+| -c, --copyright TEXT | File containing copyright data to use with output files.                                                                                                   | ../HEADER.inc |
 
-## Generate transaction builders
+> **NOTE:**
+> When no generator is given input files are parsed and checked for validity but not output files are created.
```
Comment (Jaguar0625) [inline README.md:44]: comma after given

Comment (Jaguar0625) [reply README.md:44]: not => no

File: README.md [hunk 6]
```diff
@@ -2,66 +2,76 @@
...
+>
+> See the [catbuffer-generators](https://github.com/nemtech/catbuffer-generators) project for instructions on how to use the generators.
 
-The [schemas](schemas) folder defines the entities' data structure. With the help of a code generator, you can produce the leanest code necessary to serialize and deserialize those entities.
+## Examples
 
-For example, run the following command to generate C++ transaction builders for a [TransferTransaction](https://nemtech.github.io/concepts/transfer-transaction.html#transfertransaction):
+In order to produce any output file the [catbuffer-generators](https://github.com/nemtech/catbuffer-generators) project is needed. Please see this project's usage examples.
```
Comment (Jaguar0625) [inline README.md:50]: comma after file

File: README.md [hunk 7]
```diff
@@ -2,66 +2,76 @@
...
 
-```
-python main.py --schema schemas/transfer/transfer.cats --generator cpp_builder
-```
-You can also generate code for all the schemas running the following command under the ``catbuffer-generators`` directory:
+However, ``catbuffer`` can still be used on its own to parse input files and check for their validity:
```
Comment (Jaguar0625) [inline README.md:52]: delete for

File: README.md [hunk 8]
```diff
@@ -2,66 +2,76 @@
...
+```bash
+python3 main.py --schema schemas/transfer/transfer.cats
 ```
 
-The generator creates a new file for every schema under the ``catbuffer/_generated/cpp_builder`` folder.
+There is also a script in the ``scripts`` folder to parse and validate all schemas:
 
-### Run the linter
+```bash
+scripts/generate_all.sh
```
Comment (Jaguar0625) [inline README.md:61]: you might want to mention this requires bash 5.0+

File: README.md [hunk 9]
```diff
@@ -2,66 +2,76 @@
...
 
 ### Run the tests
-```
-python -m unittest discover -v
+
+```bash
+python3 -m unittest discover -v
 ```
 
 Copyright (c) 2018 Jaguar0625, gimre, BloodyRookie, Tech Bureau, Corp Licensed under the [MIT License](LICENSE)
```
Comment (Jaguar0625) [inline README.md:77]: copyright is outdated, after fernando updates

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]
I guess you don't like commas. :joy:

File: README.md (hunk 8 above)
Comment (segfaultxavi) [reply README.md:61]: Sure? It seems to work fine with my bash 4.4

File: README.md (hunk 9 above)
Comment (segfaultxavi) [reply README.md:77]: Please advice on what should be the new Copyright notice.

**Comment by segfaultxavi** [comment]
I was under the impression commas after introductory sentences were mostly optional:
https://www.grammarly.com/blog/commas-after-introductory-phrases/
But I'll adapt to the style of the project, of course.

File: README.md (hunk 7 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (segfaultxavi) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (Jaguar0625) [reply README.md:61]: i think it was 4.0 that was required.

File: README.md (hunk 9 above)
Comment (Jaguar0625) [reply README.md:77]: https://github.com/nemtech/catbuffer/commit/bb6e1b4d8575b665aac2f251c2ea25b2be2fef4a

File: README.md (hunk 8 above)
Comment (segfaultxavi) [reply README.md:56]: Added note on Bash 4 or higher.

File: README.md (hunk 9 above)
Comment (segfaultxavi) [reply README.md]: Done

**Review by Jaguar0625 — APPROVED** [review]

**Comment by Jaguar0625** [comment]
Merged as 3b0a453.

**Pull request closed without merging**